pkg/logql/log/fmt.go

```diff
@@ -95,22 +99,29 @@ func init() {
 type LineFormatter struct {
 	*template.Template
 	buf *bytes.Buffer
+
+	currentLine []byte
 }
 
 // NewFormatter creates a new log line formatter from a given text template.
 func NewFormatter(tmpl string) (*LineFormatter, error) {
+	lf := &LineFormatter{
+		buf: bytes.NewBuffer(make([]byte, 4096)),
+	}
+	functionMap[functionLineName] = func() string {
+		return unsafeGetString(lf.currentLine)
```

Comment:
This doesn't seem thread safe 🤔

Reply:
I can add a test for the race but I think it's fine

Reply:
Well let me rephrase that, it's not thread safe but this whole code of LogQL is never run multiple line at a time.

Reply:
We can have multiple querier workers which can call this concurrently, no?

Reply:
Each worker initialized this once and execute it line by line.

Reply:
I think we're good @owen-d, I can try it in dev if that would make you more confident.

Reply:
`functionMap` is globally defined variable, so even if we have multiple workers, this looks like a problem. Am I missing something?

Reply:
Ho yeah I was focused on something like else the log line, I need to copy this map fn

Reply:
Good call it's cloned !